### ISSUE TITLE
brew cask reinstall osxfuse fails

### ISSUE THREAD

**Issue by nilbus** [Homebrew/homebrew-fuse#86]
I was unable to reinstall osxfuse cask:

```
$ brew cask reinstall osxfuse
…
==> Failed command:
/usr/bin/sudo -E -- /bin/rmdir -- #<Pathname:/usr/local/include/osxfuse>
…
```
[full output](https://gist.github.com/nilbus/9ebf88f2f5a6f49e6b66e5854ec4e7cb)

It seems that something was refactored to use a `Pathname`, and now this usage of it needs to call `#to_s`.

Should this instead be reported to the homebrew project?

FYI, after manually removing `/usr/local/include/osxfuse` and the directory that symlink points to, I was able to reinstall successfully.

```
$ brew --version ; brew cask --version
Homebrew 1.1.1
Homebrew/homebrew-core (git revision 08d0f; last commit 2016-11-21)
Homebrew-Cask 1.1.1
caskroom/homebrew-cask (git revision 30f91; last commit 2016-11-13)
```

**Comment by umireon** [comment]
`Pathname` in that context is valid.
Homebrew's `Pathname` is [monkey patched] to have `to_str` method, which enforces the Ruby to handle every Pathname is an ordinal String.
See [http://blog.bigbinary.com/2012/06/26/to_str-in-ruby.html] for it.

I doubt your `/usr/local/include/osxfuse` was a symlink, and then `rmdir` cannot remove it.
This should be leftover when osxfuse was the part of Homebrew, not of Cask.

[monkey patched]: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/pathname.rb#L329

**Comment by umireon** [comment]
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/lib/hbc/pkg.rb#L31

**Comment by nilbus** [comment]
It was a symlink, probably leftover from when osxfuse was the part of Homebrew, like you said.

Do you think this issue is invalid because of my old homebrew setup?

**Comment by umireon** [comment]
In my view, this is too trivial to report but worth to create a pull request 🍏

**Comment by nilbus** [comment]
I won't have time to dive into the code. Feel free to close this if you think it's not worth fixing.

**Comment by umireon** [comment]
I cannot close this because Im not a maintainer.

**Comment by umireon** [comment]
@nilbus This will be fixed by Homebrew/brew#1547.

**Comment by nilbus** [comment]
Thanks!
